src/main/java/jp/co/soramitsu/iroha/java/debug/LoggingTransactionStatusObserver.java

```diff
@@ -0,0 +1,80 @@
+package jp.co.soramitsu.iroha.java.debug;
+
+import iroha.protocol.Endpoint.ToriiResponse;
+import java.util.function.Consumer;
+import jp.co.soramitsu.iroha.java.TransactionStatusObserver;
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.NonNull;
+import lombok.Value;
+
+
+/**
+ * Logging observer, useful for debugging or logging.
+ *
+ * Just pass lambda, which will log string line to the constructor.
+ */
+@EqualsAndHashCode(callSuper = true)
+@Value
+@AllArgsConstructor
+public class LoggingTransactionStatusObserver extends TransactionStatusObserver {
```

Comment:
I think it'll be better to use AOP technique for logging instead of extending class. See example https://www.yegor256.com/2014/06/01/aop-aspectj-java-method-logging.html